eras/dijkstra/impl/src/Cardano/Ledger/Dijkstra/Rules/Utxo.hs

```diff
@@ -1,52 +1,296 @@
+) where
 
-import Cardano.Ledger.Allegra.Rules (shelleyToAllegraUtxoPredFailure)
+import Cardano.Ledger.Address (Addr (..))
+import Cardano.Ledger.Allegra.Rules (AllegraUtxoPredFailure, shelleyToAllegraUtxoPredFailure)
 import qualified Cardano.Ledger.Allegra.Rules as Allegra
-import Cardano.Ledger.Alonzo.Rules (AlonzoUtxoEvent, AlonzoUtxoPredFailure, AlonzoUtxosPredFailure)
+import Cardano.Ledger.Alonzo.Plutus.Context (EraPlutusContext)
+import Cardano.Ledger.Alonzo.Rules (
+  AlonzoUtxoEvent (..),
+  AlonzoUtxoPredFailure,
+  AlonzoUtxosEvent,
+  AlonzoUtxosPredFailure,
+  AlonzoUtxowEvent (..),
+ )
 import qualified Cardano.Ledger.Alonzo.Rules as Alonzo
-import Cardano.Ledger.Babbage.Rules (BabbageUtxoPredFailure)
+import Cardano.Ledger.Alonzo.UTxO (AlonzoEraUTxO, AlonzoScriptsNeeded)
+import Cardano.Ledger.Babbage.Rules (BabbageUtxoPredFailure, babbageUtxoTests)
+import Cardano.Ledger.BaseTypes (ShelleyBase, StrictMaybe (..))
+import Cardano.Ledger.Binary (DecCBOR (..), EncCBOR (..))
+import Cardano.Ledger.Binary.Coders (Decode (..), Encode (..), decode, encode, (!>), (<!))
+import Cardano.Ledger.Conway.Governance (ConwayEraGov, ConwayGovState)
 import Cardano.Ledger.Conway.Rules (
+  ConwayUTXOS,
+  ConwayUTXOW,
   ConwayUtxoPredFailure (..),
+  ConwayUtxosEvent,
   ConwayUtxosPredFailure,
+  ConwayUtxowPredFailure (..),
+  UtxoEnv,
   allegraToConwayUtxoPredFailure,
   alonzoToConwayUtxoPredFailure,
   babbageToConwayUtxoPredFailure,
  )
-import Cardano.Ledger.Dijkstra.Core (EraRuleEvent, EraRuleFailure, InjectRuleFailure (..))
-import Cardano.Ledger.Dijkstra.Era (DijkstraEra)
+import Cardano.Ledger.Credential (StakeReference (..))
+import Cardano.Ledger.Dijkstra.Core (
+  AlonzoEraTx,
+  AlonzoEraTxWits,
+  BabbageEraTxBody (..),
+  ConwayEraTxBody,
+  Era,
+  EraRule,
+  EraRuleEvent,
+  EraRuleFailure,
+  EraTx (..),
+  EraTxOut (..),
+  EraTxWits (..),
+  InjectRuleEvent (..),
+  InjectRuleFailure (..),
+  SafeToHash,
+  TxLevel (..),
+  Value,
+ )
+import Cardano.Ledger.Dijkstra.Era (DijkstraEra, DijkstraUTXO)
+import Cardano.Ledger.Dijkstra.Governance ()
 import Cardano.Ledger.Dijkstra.Rules.Utxos ()
-import Cardano.Ledger.Shelley.Rules (ShelleyUtxoPredFailure)
+import Cardano.Ledger.Dijkstra.TxBody (DijkstraEraTxBody)
+import Cardano.Ledger.Rules.ValidationMode (Test, runTest)
+import Cardano.Ledger.Shelley.LedgerState (UTxOState)
+import Cardano.Ledger.Shelley.Rules (ShelleyUtxoPredFailure, ShelleyUtxowEvent (..))
+import qualified Cardano.Ledger.Shelley.Rules as Shelley
+import Cardano.Ledger.State (EraCertState, EraGov (..), EraUTxO (..))
+import Control.DeepSeq (NFData)
+import Control.State.Transition.Extended (
+  Embed (..),
+  STS (..),
+  TRC (..),
+  TransitionRule,
+  judgmentContext,
+  trans,
+ )
+import Data.Coerce (coerce)
+import GHC.Generics (Generic)
+import Lens.Micro ((^.))
+import Validation (failure)
+
+data DijkstraUtxoPredFailure era
+  = ConwayInDijkstraUtxoPredFailure (ConwayUtxoPredFailure era)
```

Comment:
I think this wrapping of the predicate failure from the previous era is problematic  -  ( I don't remember exactly why...perhaps something to do with deserialization).

This issue seems to be about removing these kind of things from the code, if I'm not mistaken: https://github.com/IntersectMBO/cardano-ledger/issues/5299